Workiva/w_common
# CPLAT-5536: Add --watch option to compile_sass script

# [CPLAT-5536](https://jira.atl.workiva.net/browse/CPLAT-5536)
![Issue Status](http://bender.workiva.org:9000/Bender/status/CPLAT-5536)

## Motivation
The new [`compile_sass`](https://github.com/Workiva/w_common/pull/116) script does not have an option to run it as a watcher that recompiles sass files on every change.

## Changes
1. Added a `--watch` flag
2. Improved output of `--help`
3. Added a utility class to make passing around common CLI options to multiple functions less verbose
4. Added color to the output of the script
5. Compressed output file extensions can now be customized when more than one `outputStyle` is set, as long as it doesn't match the extension used for the file extension used for the "expanded" one.

#### Release Notes
Add --watch option to compile_sass script

## Review
_[See CONTRIBUTING.md][contributing-review-types] for more details on review types (+1 / QA +1 / +10) and code review process._

  

Please review: @greglittlefield-wf @corwinsheahan-wf @evanweible-wf @kealjones-wk @joebingham-wk 

### QA Checklist
        - Publink this branch to your local copy of `wdesk_sdk`
        - Run `pub run w_common:compile_sass --sourceDir lib/ --watch` from the root of your local copy of `wdesk_sdk`
        - Verify that it compiles the CSS files as expected when first starting up
        - Make a __valid__ change to any `.scss` file that starts with an `_`, and verify that it recompiles all the top-level `.scss` files.
        - Make a __valid__ change to any top level `.scss` file (one that does not start with an `_`), and verify that it only recompiles that one top-level `.scss` file.
        - Make an __invalid__ change to any `.scss` file, and verify that communicates that an error has occurred, prints the stacktrace from the sass compiler, and continues to watch for changes.
            - Revert your __invalid__ change, and verify that it once again compiles all the `.scss` files with no errors, and continues to watch for changes.

## Merge Checklist
While we perform many automated checks before auto-merging, some manual checks are needed:


[contributing-review-types]: https://github.com/Workiva/w_common/blob/master/CONTRIBUTING.md#review-types
[contributing-manual-testing]: https://github.com/Workiva/w_common/blob/master/CONTRIBUTING.md#manual-testing-criteria
